UnitTests/BraintreePayPalNativeCheckoutTests/BTPayPalNativeCheckoutClient_Tests.swift

```diff
@@ -28,6 +40,178 @@ class BTPayPalNativeCheckoutClient_Tests: XCTestCase {
+        ])
+
+        let payPalNativeCheckoutClient = BTPayPalNativeCheckoutClient(apiClient: apiClient, nativeCheckoutProvider: mockNativeCheckoutProvider)
+        let request = BTPayPalNativeCheckoutRequest(amount: "1.99")
+        payPalNativeCheckoutClient.tokenize(request) { _, _ in }
+        XCTAssertNil(mockNativeCheckoutProvider.userAuthenticationEmail)
+    }
+
+    func testTokenize_whenOnStartableApproved_returnsDidApprove() {
+        apiClient.cannedConfigurationResponseBody = BTJSON(value: [
+            "paypalEnabled": true,
+            "environment": "sandbox",
+            "paypal": ["clientId": "a-fake-client-id"] as [String: Any?]
+        ] as [String: Any])
+
+        apiClient.cannedResponseBody = BTJSON(value: [
+            "paymentResource": ["redirectUrl": "https://www.paypal.com/checkout?token=fake-ec-token"]
+        ])
+
+        let payPalNativeCheckoutClient = BTPayPalNativeCheckoutClient(apiClient: apiClient, nativeCheckoutProvider: mockNativeCheckoutProvider)
+        let request = BTPayPalNativeVaultRequest()
+        payPalNativeCheckoutClient.tokenize(request) { _, _ in }
+        mockNativeCheckoutProvider.triggerApprove(returnURL: "https://fake-return-url")
+
+        XCTAssertTrue(mockNativeCheckoutProvider.didApprove)
+    }
+
+    func testTokenize_whenOnStartableCancel_returnsCancel() {
+        apiClient.cannedConfigurationResponseBody = BTJSON(value: [
+            "paypalEnabled": true,
+            "environment": "sandbox",
+            "paypal": ["clientId": "a-fake-client-id"] as [String: Any?]
+        ] as [String: Any])
+
+        apiClient.cannedResponseBody = BTJSON(value: [
+            "paymentResource": ["redirectUrl": "https://www.paypal.com/checkout?token=fake-ec-token"]
+        ])
+
+        let payPalNativeCheckoutClient = BTPayPalNativeCheckoutClient(apiClient: apiClient, nativeCheckoutProvider: mockNativeCheckoutProvider)
+        let request = BTPayPalNativeCheckoutRequest(amount: "1.99")
+        payPalNativeCheckoutClient.tokenize(request) { _, _ in }
+        mockNativeCheckoutProvider.triggerCancel()
+        XCTAssertTrue(mockNativeCheckoutProvider.didCancel)
+
+    }
+
+    func testTokenize_whenOnStartableErrorCalled_returnsError() {
+        apiClient.cannedConfigurationResponseBody = BTJSON(value: [
+            "paypalEnabled": true,
+            "environment": "sandbox",
+            "paypal": ["clientId": "a-fake-client-id"] as [String: Any?]
+        ] as [String: Any])
+
+        let payPalNativeCheckoutClient = BTPayPalNativeCheckoutClient(apiClient: apiClient, nativeCheckoutProvider: mockNativeCheckoutProvider)
+        let request = BTPayPalNativeCheckoutRequest(amount: "1.99")
+        payPalNativeCheckoutClient.tokenize(request) { _, error in
+            self.mockNativeCheckoutProvider.triggerError(error: error as! BTPayPalNativeCheckoutError)
+        }
+
+        XCTAssertTrue(mockNativeCheckoutProvider.didError)
```

Comment:
Nit - do we want to assert on the actual error value returned instead?

Reply:
Yep- updated: https://github.com/braintree/braintree_ios/pull/1144/commits/013ff193ee4073c9dc67f9c643a2adce2cc3f993